### PR TITLE
Update CentOS Stream 9 info and container image

### PR DESCRIPTION
**Pull request by carlwgeorge**
CentOS Stream is not a rolling distribution.  It was originally advertised as one to describe how it rolls between minor versions.  However, it still has major versions and EOL dates, which by the [common definition][0] makes it not a rolling distribution.  Due the confusion, the project stopping using that term to describe the distribution.

The [EOL date for CentOS Stream 9][1] will match the date of the end of the [RHEL 9 "Full Support" phase][2], which will be 5 years after the RHEL 9 launch, which will be approximately 3 years after the RHEL 8 launch.

This also switches the container image from the stream9-development tag to the stream9 tag, which includes the ppc64le and s390x architectures.

[0]: https://en.wikipedia.org/wiki/Rolling_release
[1]: https://centos.org/centos-stream/
[2]: https://access.redhat.com/support/policy/updates/errata#Full_Support_Phase

### CODE REVIEW

**Review by Conan-Kudo — APPROVED** [review]

File: EOL.rst [hunk 1]
```diff
@@ -35,7 +35,7 @@ This mostly mimics Debian lifecycle.
 .. csv-table:: centos
    :header: "distro", "EOL", "LTS", "ELTS"
 
-   "centos stream9", "rolling", "", ""
+   "centos stream9", "~2027-05-31", "", ""
```
Comment (mayeut) [inline EOL.rst:38]:
```suggestion
   "centos stream9", "2027-05-31", "", ""
```

I guess it's probably subject to change but an isoformat string is required here to allow for date comparison later on.

File: pep600_compliance/images/os/centos.py [hunk 2]
```diff
@@ -31,9 +31,9 @@ def install_packages(self, container, machine):
 
 CENTOS_LIST: list[base.Base] = [
     CentOS(
-        "quay.io/centos/centos:stream9-development",
-        "rolling",
-        machines=["x86_64", "aarch64"],  # TODO "ppc64le", "s390x"
+        "quay.io/centos/centos:stream9",
+        ("EOL:~2027-05-31",),
```
Comment (mayeut) [inline pep600_compliance/images/os/centos.py:35]:
```suggestion
        ("EOL:2027-05-31",),
```
I guess it's probably subject to change but an isoformat string is required here to allow for date comparison later on.

**Comment by mayeut** [comment]
Thanks for clearing this up !

File: EOL.rst (hunk 1 above)
Comment (mayeut) [reply EOL.rst:38]: will be handled by automatic update.

**Pull request merged**